modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/SwiftGenerator.java

```diff
@@ -19,247 +14,248 @@
-                        "Double",
-                        "Bool",
-                        "Void",
-                        "String",
-                        "Character")
-        );
-        defaultIncludes = new HashSet<String>(
-                Arrays.asList(
-                        "NSDate",
-                        "Array",
-                        "Dictionary",
-                        "Set",
-                        "Any",
-                        "Empty",
-                        "AnyObject")
-        );
-        reservedWords = new HashSet<String>(
-                Arrays.asList(
-                        "class", "break", "as", "associativity", "deinit", "case", "dynamicType", "convenience", "enum", "continue",
-                        "false", "dynamic", "extension", "default", "is", "didSet", "func", "do", "nil", "final", "import", "else",
-                        "self", "get", "init", "fallthrough", "Self", "infix", "internal", "for", "super", "inout", "let", "if",
-                        "true", "lazy", "operator", "in", "COLUMN", "left", "private", "return", "FILE", "mutating", "protocol",
-                        "switch", "FUNCTION", "none", "public", "where", "LINE", "nonmutating", "static", "while", "optional",
-                        "struct", "override", "subscript", "postfix", "typealias", "precedence", "var", "prefix", "Protocol",
-                        "required", "right", "set", "Type", "unowned", "weak")
-        );
+    // Inject application name
+    String appName = System.getProperty("appName");
+    if (appName == null) {
+      appName = "SwaggerClient";
+    }
+    additionalProperties.put("projectName", appName);
 
-        typeMapping = new HashMap<String, String>();
-        typeMapping.put("array", "Array");
-        typeMapping.put("List", "Array");
-        typeMapping.put("map", "Dictionary");
-        typeMapping.put("date", "NSDate");
-        typeMapping.put("Date", "NSDate");
-        typeMapping.put("DateTime", "NSDate");
-        typeMapping.put("boolean", "Bool");
-        typeMapping.put("string", "String");
-        typeMapping.put("char", "Character");
-        typeMapping.put("short", "Int");
-        typeMapping.put("int", "Int");
-        typeMapping.put("long", "Int");
-        typeMapping.put("integer", "Int");
-        typeMapping.put("Integer", "Int");
-        typeMapping.put("float", "Float");
-        typeMapping.put("number", "Double");
-        typeMapping.put("double", "Double");
-        typeMapping.put("object", "AnyObject");
-        typeMapping.put("file", "NSData");
+    // Inject base url override
+    String basePathOverride = System.getProperty("basePathOverride");
+    if (basePathOverride != null) {
+      additionalProperties.put("basePathOverride", basePathOverride);
+    }
 
-        importMapping = new HashMap<String, String>();
+    // Make all the variable optional
+    String suppressRequired = System.getProperty("suppressRequired");
```

Comment:
@wing328 Hi, I've updated my pr to fix the issue above

By the way, is it bette to put new commits into a single commit? If so, I will rebase them and push again.